src/Resources/app/storefront/src/checkout/confirm-order.plugin.js

```diff
@@ -33,14 +33,38 @@ export default class ConfirmOrderPlugin extends Plugin {
 
     init() {
         this._client = new StoreApiClient();
+        this.selectedAdyenPaymentMethod = this.getSelectedPaymentMethodKey();
         this.confirmOrderForm = DomAccess.querySelector(document, '#confirmOrderForm');
         this.confirmFormSubmit = DomAccess.querySelector(document, '#confirmOrderForm button[type="submit"]');
-        this.confirmFormSubmit.addEventListener('click', this.onConfirmOrderSubmit.bind(this));
-        this.paymentComponent = $(`[data-adyen-payment-component]`);
         this.responseHandler = this.handlePaymentAction;
         this.adyenCheckout = Promise;
         this.initializeCheckoutComponent().then(function () {
-            this.initializeCustomPayButton();
+
+            // Non adyen payment method selected
+            // this can not happen, because this js plugin is registered only if adyen methods selected
+            // PluginManager.register('ConfirmOrderPlugin', ConfirmOrderPlugin, '#adyen-payment-checkout-mask');
+            if (adyenCheckoutOptions.selectedPaymentMethodPluginId !==
+                adyenCheckoutOptions.adyenPluginId) {
+                return;
+            }
+
+            if (!adyenCheckoutOptions || !adyenCheckoutOptions.paymentStatusUrl ||
+                !adyenCheckoutOptions.checkoutOrderUrl || !adyenCheckoutOptions.paymentHandleUrl) {
+                console.error('Adyen payment configuration missing.');
+                return;
+            }
+
+            if (this.selectedAdyenPaymentMethod in adyenConfiguration.componentsWithPayButton) {
+                // replaces confirm button with adyen pay button for paywithgoogle, applepay etc.
+                this.initializeCustomPayButton();
+            }
+
+            if (adyenConfiguration.updatablePaymentMethods.includes(this.selectedAdyenPaymentMethod) && !this.stateData) {
```

Comment:
I think the `&& !this.stateData` part is not strictly necessary, since the page refreshes when a new payment method is selected, this part will always be `true`

Reply:
maybe this could be useful if Shopware ever switches back to NOT refreshing the page after every PM selection. For now though, the whole concept of updatable payment methods is obsolete.